openqabot/approver.py

```diff
@@ -131,6 +132,53 @@ def is_job_marked_acceptable_for_incident(self, job_id: int, inc: int) -> bool:
             pass
         return False
 
+    @lru_cache(maxsize=512)
+    def was_green_before(self, job_id: int, inc: int) -> bool:
+        log.info("Current aggregate test %s failed. Will look at older jobs" % job_id)
+        # We need a considerable amount of older jobs, since there could be many failed manual restarts from same day
+        older_jobs = self.client.get_older_jobs(job_id, 20)
+
+        current_build = older_jobs['data'][0]['build'].replace('-1','')
+        log.info(current_build)
+        current_build_date = date(
+            int(current_build[0:4]),
+            int(current_build[4:6]),
+            int(current_build[6:8])
+            )
+        log.info("Failed job date: %s",current_build_date)
+
+        # Use at most X days old build. Don't go back in time too much to reduce risk of using invalid tests
+        oldest_build_usable = current_build_date - timedelta(days=6)
+
+        regex = re.compile(r"(.*)Maintenance:/%s(.*)" % inc)
+        # Skipping first job, which was the current one
+        for i in range(1, len(older_jobs['data'])):
+            job = older_jobs['data'][i]
+            job_build = job['build'].replace('-1','')
+            job_build_date = date(
+                int(job_build[0:4]),
+                int(job_build[4:6]),
+                int(job_build[6:8])
+                )
+            # Check the job is not too old
+            if job_build_date > oldest_build_usable:
+                if job['result'] == 'passed' or job['result'] == 'softfailed':
+                    # Check the job contains the update under test
+                    job_settings = self.client.get_single_job(job['id'])
+                    if not regex.match(str(job_settings)):
+                        # Likely older jobs don't have it either. Giving up
+                        log.info("Not found an older passing job with update %s under test present. Giving up" % inc)
+                        return False
+
+                    log.info("Found a valid job result %s for update %s" % (job['id'], inc))
+                    return True
+            else:
+                log.info("Job is too old. Giving up")
+                return False
+        # Run out of jobs. Giving up
+        log.info("Run out of jobs to try. Giving up")
```

Comment:
The comment is redundant due to the log message in the next line. I also often prefer having log messages over comments.

```suggestion
        log.info("Ran out of jobs to try. Giving up")
```